stackstac/rio_reader.py

```diff
@@ -357,6 +357,17 @@ def _open(self) -> ThreadsafeRioDataset:
                         resampling=self.resampling,
                         **self.spec.vrt_params,
                     )
+            # Alternatively, try to make VRT when ground control points (gcps) are present
+            elif ds.gcps is not None:
+                with self.gdal_env.open_vrt:
+                    src_crs = ds.gcps[-1]
+                    vrt = WarpedVRT(
+                        ds,
+                        src_crs=src_crs,
```

Comment:
nit: Could we refactor things a little here to avoid copy-pasting this `WarpedVRT` instantiation?

I'm thinking maybe something like

```python
if ds.crs is not None and self.spec.vrt_params == {"crs": ds.crs.to_epsg(), ...}:
    logger.info(f"Skipping VRT for {self.url!r}")
    vrt = None
else:
    # Need to warp
    if ds.crs is not None:
        src_crs = ds.crs
    elif ds.gcps is not None:
        src_crs = ds.gcps[-1]
    else:
        raise ValueError(f"{self.url!r} does not have georeferencing!") # add more info
    vrt = WarpedVRT(ds, src_crs=src_crs, ...)
```

Reply:
"Flat is better than nested" :wink: Agree though that it's possible to have just one `WarpedVRT` call, have refactored it in 1b3c47c6bd0d3ec9dd17b25dc8cb1c8496a131a7, see if it looks ok.